swarm/storage/hasherstore.go

```diff
@@ -121,31 +135,42 @@ func (h *hasherStore) Close() {
 // Wait returns when
 //    1) the Close() function has been called and
 //    2) all the chunks which has been Put has been stored
+//    OR
+//    1) if there is error while storing chunk
 func (h *hasherStore) Wait(ctx context.Context) error {
 	defer close(h.quitC)
+	select {
+	case err := <-h.waitC:
+		return err
+	}
+}
+
+
+func (h *hasherStore) startWait(ctx context.Context)  {
```

Comment:
i prefer returning the error and push it onto the waitC where it is called

Reply:
ooh.. thats a mistake from my side.. Will fix it